packages/ember-application/lib/system/resolver.js

```diff
@@ -217,6 +217,10 @@ export default EmberObject.extend({
     let dirname = lastSlashIndex !== -1 ? name.slice(0, lastSlashIndex) : null;
 
     if (type !== 'template' && lastSlashIndex !== -1) {
+      deprecate(`Using / as a namespace, used by: "${name}"`, true, {
+        id: 'slash-as-namespace',
+        until: 'next LTS' // TODO
```

Comment:
Whatever the next LTS is. At this point, `2.16.0`.

This should also have a test.

ref: https://emberjs.com/api/classes/Ember.html#method_deprecate